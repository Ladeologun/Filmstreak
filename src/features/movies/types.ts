import { ImageSourcePropType } from "react-native"

export interface Movie {
    id:number
    backPoster?: string
    popularity:number
    title:string
    poster: string
    overview?: string
    rating:number
    language:string
    video?:boolean
    date:string
}


export interface TMDBMovie {
    id:number
    backdrop_path?: string
    popularity:number
    title:string
    poster_path?: string
    overview?: string
    vote_average:number
    original_language:string
    video?:boolean
    release_date: string
}

export interface NavigationTypes {
    navigate: (name: string, params?: { [key: string]: any }) => void;
    replace?: (name: string, params?: { [key: string]: any }) => void;
    canGoBack?: () => boolean;
    goBack: () => void;
}
