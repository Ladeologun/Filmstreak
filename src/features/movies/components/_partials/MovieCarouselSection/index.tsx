
import React from 'react';
import { View,FlatList } from 'react-native';
import { CategoryHeader, ListItemSeperator, MovieCard } from '~components';
import { Movie } from "~movies/types";

interface Iprops{
    movies:Movie[];
    title?: string;
    showRightCTA?:boolean
}

const MovieCarouselSection: React.FC<Iprops> =({movies, title="Popular", showRightCTA=false})=> {
    
    return (<>
        {<View>
            <CategoryHeader title={title} showRightCTA={showRightCTA}/>
            <FlatList
                data={movies}
                keyExtractor={item => item.id.toString()}
                key={'_'}
                horizontal
                renderItem={({item})=><MovieCard item={item} section={title}/>}
                ItemSeparatorComponent={ListItemSeperator}
            />
        </View>}
        </>
    );
}


export default MovieCarouselSection;