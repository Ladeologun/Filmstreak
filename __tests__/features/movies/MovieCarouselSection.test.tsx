import renderer from 'react-test-renderer';
import {test,expect} from '@jest/globals';
import MovieCarouselSection from '~movies/components/_partials/MovieCarouselSection';
import { Movie } from '~movies/types';


const MovieBuilder:Movie = {
    id: 1,
    backPoster: "https://example.com/backposter.jpg",
    popularity: 85.2,
    title: "The Great Adventure",
    poster: "https://example.com/poster.jpg",
    overview: "A thrilling adventure of a lifetime.",
    rating: 8.5,
    language: "English",
    video: true,
    date: "2024-04-12"
};

test('renders correctly', () => {
    const tree = renderer.create(<MovieCarouselSection movies={[MovieBuilder]} />).toJSON();
    expect(tree).toMatchSnapshot();
});
