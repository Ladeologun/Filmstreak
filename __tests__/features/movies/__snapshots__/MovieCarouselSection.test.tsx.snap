// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginBottom": 20,
        "paddingBottom": 5,
        "paddingTop": 15,
      }
    }
  >
    <Text
      style={
        {
          "borderBottomColor": "green",
          "color": "#ffffff",
          "fontFamily": "PermanentMarker-Regular",
          "fontSize": 19,
          "fontWeight": "800",
          "textDecorationColor": "gray",
          "textShadowColor": "gray",
          "textShadowOffset": {
            "height": 0.5,
            "width": 0.5,
          },
        }
      }
    >
      Popular
    </Text>
  </View>
  <RCTScrollView
    ItemSeparatorComponent={[Function]}
    data={
      [
        {
          "backPoster": "https://example.com/backposter.jpg",
          "date": "2024-04-12",
          "id": 1,
          "language": "English",
          "overview": "A thrilling adventure of a lifetime.",
          "popularity": 85.2,
          "poster": "https://example.com/poster.jpg",
          "rating": 8.5,
          "title": "The Great Adventure",
          "video": true,
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={true}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={0.0001}
    stickyHeaderIndices={[]}
    viewabilityConfigCallbackPairs={[]}
  >
    <View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={
          [
            {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "marginBottom": 10,
              "width": 375,
            }
          }
        >
          <View
            accessibilityIgnoresInvertColors={true}
            style={
              {
                "alignItems": "center",
                "borderRadius": 10,
                "height": 150,
                "justifyContent": "center",
                "marginBottom": 10,
                "overflow": "hidden",
                "width": "100%",
              }
            }
          >
            <Image
              source={
                {
                  "uri": "https://example.com/poster.jpg",
                }
              }
              style={
                [
                  {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                  {
                    "height": 150,
                    "width": "100%",
                  },
                  undefined,
                ]
              }
            />
            <Image
              source={
                {
                  "testUri": "../../../src/assets/drawables/play-button.png",
                }
              }
              style={
                {
                  "height": 45,
                  "tintColor": "skyblue",
                  "width": 45,
                }
              }
            />
          </View>
          <Text
            numberOfLines={1}
            style={
              {
                "color": "#ffffff",
                "fontFamily": "Raleway-BoldItalic",
                "fontSize": 16,
              }
            }
          >
             
            The Great Adventure
             
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
